tests/datasets/test_dataloader.py

```diff
@@ -263,3 +263,39 @@ def test_subindicators_update(self, dataset, datasetData, subindicatorGroup):
         assert len(groups) == 2
         assert groups[0].subindicators == ["A", "B", "C"]
         assert groups[1].subindicators == ["X", "Y"]
+
+    def test_new_dataset_same_group_name(self, dataset, datasetData, subindicatorGroup):
+        # Make sure datasets with new subindicators are not leaking into groups with the same name from previous datasets
+        new_dataset = DatasetFactory()
+
+        DatasetDataFactory(
+            dataset=new_dataset, data={
+                'group1': 'D', 'group2': 'Z'
+            }
+        )
+
+        groups = dataloader.create_groups(new_dataset, ["group1", "group2"])
+
+        assert len(groups) == 2
+        assert groups[0].subindicators == ["D"]
+        assert groups[0].subindicators != ["A", "B", "C", "D"]
+        assert groups[1].subindicators == ["Z"]
+        assert groups[1].subindicators != ["X", "Y", "Z"]
+
+    def test_old_dataset_does_not_include_new_subindicators(self, dataset, datasetData, subindicatorGroup):
+        # Make sure same group names from a previous dataset does not have subindicators leaking into it
+
+        DatasetDataFactory(
```

Comment:
this is not the right test, this does not create a new dataset.

You should do the same as in above test:
```
new_dataset = DatasetFactory()
DatasetDataFactory(dataset=new_dataset, data={'group1': 'D', 'group2': 'Z' } )
```
otherwise D and Z will never be there and cannot leak into the old group with the same name.